mart/attack/adversary.py

```diff
@@ -114,13 +118,9 @@ def configure_optimizers(self):
     def training_step(self, batch_and_model, batch_idx):
         input, target, model = batch_and_model
 
-        # Compose input_adv from input, then give to model for updated gain.
-        perturbation = self.perturber(input=input, target=target)
-        input_adv = self.composer(perturbation, input=input, target=target)
-
         # Target model expects input in the original format.
-        # TODO: We assume batch is a tuple, but it may be different for models outside MART. Add a reverse transform?
-        batch_adv = (input_adv, target)
+        # We just compose adversarial examples but not enforce the threat model until after fit().
+        batch_adv = self.forward(input=input, target=target, enforce=False)
```

Comment:
This can become `self.forward(batch, enforce=False)` if you make the changes described.

Reply:
Fixed.